src/index.js

```diff
@@ -0,0 +1,133 @@
+import './style.css';
+
+import {
+  dateElement,
+  options,
+  today,
+} from './modules/date.js';
+import renderTodo, {
+  list,
+} from './modules/status.js';
+
+dateElement.innerHTML = today.toLocaleDateString('en-US', options);
+const clearAllBtn = document.querySelector('.clear-todos');
+let userInput = localStorage.getItem('todoItemsRef') ?
+  JSON.parse(localStorage.getItem('todoItemsRef')) : [];
+
+function addTodo(text) {
+  const todo = {
+    listItem: text,
+    completed: false,
+    id: Date.now(),
+  };
+
+  userInput.push(todo);
+  renderTodo(userInput);
+}
+
+function deleteTodo(key) {
+  const li = key.parentElement.parentElement;
+  const title = li.querySelector('.task').textContent;
+  userInput = userInput.filter((item) => item.listItem !== title);
+
+  localStorage.setItem('todoItemsRef', JSON.stringify(userInput));
```

Comment:
Since this line of code is repeated many times, you could use a function for it like so : function LocalStorageAdd() {
  return localStorage.setItem('todoItemsRef', JSON.stringify(userInput));}

Reply:
Thank you. This solution worked. However, when I refreshed the application page, local storage gave unexpected behavior. On deleting function after adding i.e (1, 1, 1, 1, ) deleting one item, erased all the local storage. I am not clear on the bug